mpas_analysis/test/test_timekeeping.py

```diff
@@ -0,0 +1,200 @@
+                                  second=1)
+        self.assertEqual(date1, date2)
+
+        # hh.mm.ss
+        calendar = 'gregorian'
+        date1 = stringToRelativedelta('00.00.01',
+                                      calendar=calendar)
+        date2 = MpasRelativeDelta(years=0, months=0, days=0, hours=0,
+                                  minutes=0, seconds=1, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        # YYYY-MM-DD
+        date1 = stringToDatetime('0001-01-01')
+        date2 = datetime.datetime(year=1, month=1, day=1, hour=0, minute=0,
+                                  second=0)
+        self.assertEqual(date1, date2)
+
+        # SSSSS
+        date1 = stringToRelativedelta('00005',
+                                      calendar=calendar)
+        date2 = MpasRelativeDelta(years=0, months=0, days=0, hours=0,
+                                  minutes=0, seconds=5, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0005-00-00',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1991-01-15'))
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0000-02-00',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1995-11-15'))
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0000-00-20',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1995-12-26'))
+
+        # since pandas and xarray use the numpy type 'datetime[ns]`, which
+        # has a limited range of dates, the date 0001-01-01 gets increased to
+        # the minimum allowed year boundary, 1678-01-01 to avoid invalid
+        # dates.
+        date1 = clampToNumpyDatetime64(stringToDatetime('0001-01-01'),
+                                       yearOffset=0)
+        date2 = datetime.datetime(year=1678, month=1, day=1)
+        self.assertEqual(date1, date2)
+
+        date1 = clampToNumpyDatetime64(stringToDatetime('0001-01-01'),
+                                       yearOffset=1849)
+        date2 = datetime.datetime(year=1850, month=1, day=1)
+        self.assertEqual(date1, date2)
+
+        # since pandas and xarray use the numpy type 'datetime[ns]`, which
+        # has a limited range of dates, the date 9999-01-01 gets decreased to
+        # the maximum allowed year boundary, 2262-01-01 to avoid invalid
+        # dates.
+        date1 = clampToNumpyDatetime64(stringToDatetime('9999-01-01'),
+                                       yearOffset=0)
+        date2 = datetime.datetime(year=2262, month=1, day=1)
+        self.assertEqual(date1, date2)
+
+        # test if the calendars behave as they should
```

Comment:
I don't mind doing this and I agree it makes sense to systematically test both calendars but there will be no large reduction in code.  Each test is unique (except for the first 2) independent of calendar.